File: minegame-database/src/main/java/top/nipuru/minegame/database/processor/SaveFileHandler.java
package top.nipuru.minegame.database.processor;

import lombok.AllArgsConstructor;
import top.nipuru.minegame.common.message.database.SaveFileRequest;
import top.nipuru.minegame.common.processor.RequestDispatcher;
import top.nipuru.minegame.database.DatabaseServer;

@AllArgsConstructor
public class SaveFileHandler implements RequestDispatcher.Handler<SaveFileRequest> {

    private final DatabaseServer server;

    @Override
    public void handle(RequestDispatcher.ResponseSender responseSender, SaveFileRequest request) throws Exception {
        server.getFileManager().saveFile(request.getFilename(), request.getData());
        responseSender.sendResponse(true); // response
    }

    @Override
    public Class<SaveFileRequest> interest() {
        return SaveFileRequest.class;
    }
}

File: minegame-database/src/main/kotlin/top/nipuru/minegame/database/processor/SaveFileHandler.kt
package top.nipuru.minegame.database.processor

import com.alipay.remoting.AsyncContext
import top.nipuru.minegame.common.message.database.SaveFileRequest
import top.nipuru.minegame.common.processor.RequestDispatcher
import top.nipuru.minegame.database.file.FileManager

class SaveFileHandler : RequestDispatcher.Handler<SaveFileRequest> {

    override fun handle(asyncCtx: AsyncContext, request: SaveFileRequest) {
        FileManager.saveFile(request.filename, request.data)
        asyncCtx.sendResponse(true) // response
    }

    override fun interest(): Class<SaveFileRequest> {
        return SaveFileRequest::class.java
    }
}
